File: Assets/Scripts/Network/CommonStructures/NetworkPlayerSimulationState.cs
namespace Network
{
    /// <summary>
    /// The simulation state of a player. (Result of a input state)
    /// </summary>
    public class NetworkPlayerSimulationState
    {
        /// <summary>
        /// The simulation frame of the player.
        /// </summary>
        public int simulationFrame;
        /// <summary>
        /// The direction of the player.
        /// </summary>
        public float direction;
        /// <summary>
        /// The position of the player.
        /// </summary>
        public Position position;
    }
}

File: Server/src/commonStructures/networkPlayerSimulationState.ts
import Position from "./position";

/**
 * The simulation state of a player. (Result of a input state)
 */
export default class NetworkPlayerSimulationState {
    /**
     * The simulation frame of the player.
     */
    public simulationFrame : number = 0;
    /**
     * The direction of the player.
     */
    public direction:number = 0;
    /** 
     * The position of the player.
     */
    public position : Position = new Position(0,0);
}